oasislmf/gen_test_data/generate_test_data.py

```diff
@@ -0,0 +1,725 @@
+import argparse
+from collections import OrderedDict
+import numpy as np
+import os
+import struct
+
+
+class ModelFile:
+    def __init__(self):
+        pass
+
+    def seed_rng(self):
+        if self.random_seed == 0:
+            np.random.seed(0)
+        elif self.random_seed == -1:
+            np.random.seed(1234)
+        else:
+            np.random.seed(self.random_seed)
+
+    def write_file(self):
+        with open(self.file_name, 'wb') as f:
+            if self.start_stats:
+                for k, v in self.start_stats.items():
+                    f.write(struct.pack(v['dtype'], v['value']))
+            for col in self.data:
+                for idx, dtype in enumerate(self.dtypes.values()):
+                    try:
+                        f.write(struct.pack(dtype, col[idx]))
+                    except TypeError:   # If line is not tuple
+                        f.write(struct.pack(dtype, col))
```

Comment:
you can use struct like this
>>> col = (1,2,3)
>>> struct.pack('idi', *col)
b'\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00@\x03\x00\x00\x00'
you don't need to iterate
I would also enforce self.data to yield tuples to remove the usage of the try except